product-scenarios/scenarios-common/src/main/java/org/wso2/am/scenario/test/common/ScenarioTestBase.java

```diff
@@ -261,6 +262,23 @@ public void createRole(String adminUsername, String adminPassword, String role,
 
     }
 
+    public void updateRole(String adminUsername, String adminPassword, String role, String[] userList,
+                           String[] permissionArray) throws APIManagementException {
+
+        UserManagementClient userManagementClient = null;
+        try {
+            userManagementClient = getRemoteUserManagerClient(adminUsername, adminPassword);
+            userManagementClient.deleteRole(role);
+            userManagementClient.addRole(role,
```

Comment:
Is the permissionArray contains (existingPermissions + newPermissions - removingPermissions)?

Reply:
We remove the existing permissions from the role which assigned for a user and replaces it with the new permission